lib/device/fan.js

```diff
@@ -87,6 +98,73 @@ export default class {
     }
   }
 
+  async internalSwingUpdate(value) {
+    try {
+      const newValue = value ? 'SWING_ON' : 'SWING_OFF';
+      // Don't continue if the new value is the same as before
+      if (this.cacheSwing === value) {
+        return;
+      }
+
+      await this.platform.sendDeviceUpdate(this.accessory, {
+        cmd: 'stateFan',
+        value: fanState[newValue],
+      });
+
+      // Cache the new state and log if appropriate
+      this.cacheSwing = newValue;
+      if (this.enableLogging) {
+        this.log('[%s] %s [%s].', this.name, platformLang.curState, newValue);
+      }
+    } catch (err) {
+      // Catch any errors during the process
+      const eText = parseError(err);
+      this.log.warn('[%s] %s %s.', this.name, platformLang.devNotUpdated, eText);
+
+      // Throw a 'no response' error and set a timeout to revert this after 2 seconds
+      setTimeout(() => {
+        this.service.updateCharacteristic(this.hapChar.On, this.cacheState === 'on');
+      }, 2000);
+      throw new this.hapErr(-70402);
+    }
+  }
+
+  async internalSpeedUpdate(value) {
+    try {
+      // Don't continue if the value is lower than 12.5
+      if (value < 12.5) {
+        return;
+      }
+      const newValue = `SPEED_${value * 0.08}`;
+      // Don't continue if the new value is the same as before
+      if (this.cacheSpeed === value) {
+        return;
+      }
+
+      await this.platform.sendDeviceUpdate(this.accessory, {
+        cmd: 'stateFan',
+        value: fanState[newValue],
+      });
+
+      // Cache the new state and log if appropriate
+      this.cacheSpeed = newValue;
+      if (this.enableLogging) {
+        this.log('[%s] %s [%s].', this.name, platformLang.curState, newValue);
+      }
+    } catch (err) {
+      // Catch any errors during the process
+      const eText = parseError(err);
+      this.log.warn('[%s] %s %s.', this.name, platformLang.devNotUpdated, eText);
+
+      // Throw a 'no response' error and set a timeout to revert this after 2 seconds
+      setTimeout(() => {
+        this.service.updateCharacteristic(this.hapChar.On, this.cacheState === 'on');
```

Comment:
Same here with `this.hapChar.RotationSpeed` and use the previous cache value for the value.